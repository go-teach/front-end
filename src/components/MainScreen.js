import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import {
	Button,
	StyleSheet,
	Text,
	View,
	TextInput,
} from 'react-native';
import { NavigationActions } from 'react-navigation';
import { loggingIn } from '../actions/Auth';

const styles = StyleSheet.create({
	container: {
		flex: 1,
		justifyContent: 'center',
		alignItems: 'center',
	},
	welcome: {
		fontSize: 20,
		textAlign: 'center',
		margin: 10,
	},
	inputBox: {
		height: 35,
		width: 200,
		backgroundColor: 'white',
		borderColor: 'black',
		borderWidth: 1,
		margin: 10,
		padding: 3,
	},
});

class MainScreen extends Component {
	static propTypes = {
		navToRegister: PropTypes.func.isRequired,
		navToProfile: PropTypes.func.isRequired,
	}

	static navigationOptions = {
		title: 'Log In',
	}

	constructor(props) {
		super(props);
		this.state = {
			username: '',
			password: '',
		};

		this.handleLogIn = this.handleLogIn.bind(this);
	}

	handleLogIn() {
		const { navToProfile, loggingIn } = this.props;
		loggingIn();
		navToProfile();
	}

	render() {
		const { navToRegister, navToProfile } = this.props;

		return (
			<View style={styles.container}>
				<Text style={styles.welcome}>Go-Teach </Text>
				<TextInput
					underlineColorAndroid="transparent"
					style={styles.inputBox}
					onChangeText={text => this.setState({ username: text })}
					placeholder="Username"
				/>
				<TextInput
					underlineColorAndroid="transparent"
					secureTextEntry
					style={styles.inputBox}
					onChangeText={text => this.setState({ password: text })}
					placeholder="Password"
				/>
				<View style={{ flexDirection: 'row' }}>
					<Button
						onPress={this.handleLogIn}
						title="Log in"
					/>
					<Text> or </Text>
					<Text style={{ color: 'blue' }} onPress={navToRegister}> Sign Up </Text>
				</View>
			</View>
		);
	}
}

const mapDispatchToProps = dispatch => ({
	navToRegister: () => dispatch(NavigationActions.navigate({ routeName: 'Register' })),
	navToProfile: () => dispatch(NavigationActions.navigate({ routeName: 'Profile' })),
	loggingIn: () => dispatch(loggingIn()),
});

export { MainScreen }; //for testing
export default connect(undefined, mapDispatchToProps)(MainScreen);
