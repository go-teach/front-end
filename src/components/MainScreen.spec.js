import React from 'react';
import { shallow } from 'enzyme';
import { MainScreen } from './MainScreen';

describe('MainScreen component', () => {
	it('should return proper component', () => {
		const mockedFunc1 = jest.fn();
		const mockedFunc2 = jest.fn();
		const wrapper = shallow(<MainScreen navToRegister={mockedFunc1} navToProfile={mockedFunc2} />);
		expect(wrapper).toMatchSnapshot();
	});
});
