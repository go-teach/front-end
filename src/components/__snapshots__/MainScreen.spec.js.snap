// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainScreen component should return proper component 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }
  >
    Go-Teach 
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Username"
    style={
      Object {
        "backgroundColor": "white",
        "borderColor": "black",
        "borderWidth": 1,
        "height": 35,
        "margin": 10,
        "padding": 3,
        "width": 200,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Password"
    secureTextEntry={true}
    style={
      Object {
        "backgroundColor": "white",
        "borderColor": "black",
        "borderWidth": 1,
        "height": 35,
        "margin": 10,
        "padding": 3,
        "width": 200,
      }
    }
    underlineColorAndroid="transparent"
  />
  <Component
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Button
      onPress={[Function]}
      title="Log in"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
       or 
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      onPress={[MockFunction]}
      style={
        Object {
          "color": "blue",
        }
      }
    >
       Sign Up 
    </Text>
  </Component>
</Component>
`;
